lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadataInfo.php

```
@@ -468,6 +469,29 @@ public function __construct($documentName)
     }
 
     /**
+     * Helper method to get reference id of ref* type references
+     * @param array  $reference
+     * @param string $storeAs
+     * @return mixed
+     * @internal
+     */
+    public static function getReferenceId($reference, $storeAs)
+    {
+        return $reference[ClassMetadataInfo::getReferencePrefix($storeAs) . 'id'];
+    }
+
+    /**
+     * Returns the reference prefix used for a referene
+     * @param string $storeAs
+     * @return string
+     * @internal
+     */
+    public static function getReferencePrefix($storeAs)
+    {
+        return $storeAs === ClassMetadataInfo::REFERENCE_STORE_AS_REF ? '' : '$';
```

I assume "$" only makes sense for one of the two DBRef reference types. Would you want that returned if this is somehow called for `REFERENCE_STORE_AS_ID`? Alternatively, if this method should never be called in that case perhaps we should throw a logic exception.

---

Agree on the exception - neither prefix is valid for `REFERENCE_STORE_AS_ID`.